tests/unit/s2n_build_test.c

```diff
@@ -86,8 +86,6 @@ S2N_RESULT s2n_check_supported_libcrypto(const char *s2n_libcrypto)
         { .libcrypto = "openssl-1.1.1", .is_openssl = true },
         { .libcrypto = "openssl-3.0", .is_openssl = true },
         { .libcrypto = "openssl-3.0-fips", .is_openssl = true, .is_opensslfips = true },
-        { .libcrypto = "openssl-3.2.2", .is_openssl = true },
-        { .libcrypto = "openssl-3.4", .is_openssl = true },
```

Comment:
Don't we still need this? the original PR for for al2023 https://github.com/aws/s2n-tls/pull/5255

Reply:
No. I reverted that PR and fixed it differently. We should not be setting S2N_LIBCRYPTO to any system default openssl, which is what those are. The openssl-3.4 was coming from MacOS, and the openssl-3.2.2 was coming from AL2023. Now MacOS doesn't set S2N_LIBCRYPTO at all and AL2023 sets "default".